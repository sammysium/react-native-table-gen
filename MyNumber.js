import PropTypes from 'prop-types';
import React from 'react';
import { StyleSheet, Text } from 'react-native';
import { formatNumber }  from './helperFunctions';

const MyComponent = ({
  value,
  decimalsToShow,
  styles,
  rest
}) => {
  const txtStyle = styles !== undefined ? styles : defaultStyles.text;
  const output = formatNumber(
    value,
    decimalsToShow === undefined ? 2 : decimalsToShow
  );

  return (
    <>
      <Text style={txtStyle} {...rest}>{output}</Text>
    </>
  );
};

const defaultStyles = StyleSheet.create({
  text: {
    color: '#000',
    marginBottom: 4,
    marginRight: 10
  }
});
MyComponent.propTypes = {
  value: PropTypes.number.isRequired,
  decimalsToShow: PropTypes.number,
  styles: PropTypes.instanceOf(Object),
  rest: PropTypes.instanceOf(Object)
};
MyComponent.defaultProps ={
  styles: undefined,
  decimalsToShow: 0,
  rest: {}
};
const MyNumber = React.memo(MyComponent);
export default MyNumber;
