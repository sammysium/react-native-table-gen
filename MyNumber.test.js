import React from 'react';
import { Text } from 'react-native';
import { render , fireEvent } from 'react-native-testing-library';
import MyNumber from './MyNumber';

describe('component MyNumber', () => {
  const value = 2989;
  it('renders correctly', () => {
    const snapShotTree = render(<MyNumber value={value} decimalsToShow={2} />).toJSON();
    expect(snapShotTree).toMatchSnapshot();
  });

  
  it('should return 2 decimal number', () => {
    const App = () => (
      <>
        <MyNumber value={value} decimalsToShow={2} />
      </>
    );

    const wrapper = render(<App />);
    const label = wrapper.UNSAFE_getByType(Text);
    const lbl = `2,989.00`;
    expect(label.props.children).toStrictEqual(lbl);
  });

   
  it('should return 0 decimal number', () => {
    const App = () => (
      <>
        <MyNumber value={value} decimalsToShow={0} />
      </>
    );

    const wrapper = render(<App />);
    const label = wrapper.UNSAFE_getByType(Text);
    const lbl = `2989`;
    expect(label.props.children).toStrictEqual(lbl);
  });

     
  it('should return rounded 0 decimal number', () => {
    const App = () => (
      <>
        <MyNumber value={350.9889} decimalsToShow={0} />
      </>
    );

    const wrapper = render(<App />);
    const label = wrapper.UNSAFE_getByType(Text);
    const lbl = `351`;
    expect(label.props.children).toStrictEqual(lbl);
  });

       
  it('should return rounded 2 decimal number', () => {
    const App = () => (
      <>
        <MyNumber value={350.9889} decimalsToShow={2} />
      </>
    );

    const wrapper = render(<App />);
    const label = wrapper.UNSAFE_getByType(Text);
    const lbl = `350.99`;
    expect(label.props.children).toStrictEqual(lbl);
  });

  it('should be pressed ', () => {
    const events = {
      onPress: jest.fn(()=>{
        console.log('pressed');
      })
    }
    const App = () => (
      <>
        <MyNumber value={350.9889} {...events} decimalsToShow={2} />
      </>
    );
      const spy = jest.spyOn(console,'log')
    const wrapper = render(<App />);
    const label = wrapper.UNSAFE_getByType(Text);
    fireEvent.press(label)
    expect(spy).toHaveBeenCalled();
  });

});
