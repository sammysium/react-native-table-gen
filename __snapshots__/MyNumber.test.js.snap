// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component MyNumber renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "#000",
      "marginBottom": 4,
      "marginRight": 10,
    }
  }
>
  2,989.00
</Text>
`;
